### PR TITLE
Homework3: Dmitry Subbotin

### PR DESCRIPTION
**Pull request by DmitrySub**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework3/task4.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def combinations_without_repetitions(n: int, k: int) -> int:
+    """Число сочетаний без повторений из n по к
+
+    Генерирует исключение ValueError, если n < k.
+    """
+    if n < k:
+        raise ValueError('n < k')
+    return int(factorial(n) / (factorial(n - k) * factorial(k)))
```
Comment (belovmd) [inline src/homework3/task4.py]: O(n!) complexity. has O(1) solution

**Review by belovmd — CHANGES_REQUESTED** [review]
great job. 
please address this minor issue

File: src/homework3/task4.py (hunk 1 above)
Comment (DmitrySub) [reply src/homework3/task4.py]: So far I am at a loss with the solution of O(1).
How to make the O(1) if we need to bypass the input list at least once? => O(n)

Comment (belovmd) [reply src/homework3/task4.py]: it can be O(1) instead of O(n!) for counting - just reworked combinations. we surely have O(n) for read

**Review by belovmd — APPROVED** [review]
9

**Pull request closed without merging**